src/pamiq_core/model/interface.py

```diff
@@ -0,0 +1,115 @@
+from abc import ABC, abstractmethod
+from typing import Any
+
+
+class InferenceModel(ABC):
+    """Wrapper class for model to infer.
+
+    Needed for multi-thread training and inference in parallel.
+    """
+
+    @abstractmethod
+    def infer(self, *args, **kwds) -> Any:
+        """Perform inference using a model.
+
+        Args:
+            *args:
+                Positional arguments required for inference.
+            **kwds:
+                Keyword arguments required for inference.
+        Returns:
+            Any:
+                The result of the inference.
+        """
+        raise NotImplementedError
+
+    def __call__(self, *args, **kwds) -> Any:
+        """Perform inference using a model.
+
+        Args:
+            *args:
+                Positional arguments required for inference.
+            **kwds:
+                Keyword arguments required for inference.
+        Returns:
+            Any:
+                The result of the inference.
+        """
+        return self.infer(*args, **kwds)
+
+
+class TrainingModel(ABC):
+    """Wrapper class to train model.
+
+    Needed for multi-thread training and inference in parallel.
+    """
+
+    def __init__(self, has_inference_model: bool = True, inference_only: bool = False):
+        """Initialize.
+
+        Args:
+            has_inference_model (bool):
+                Whether to have InferenceModel.
+                Default to True.
+            inference_only (bool):
+                Whether to do Inference only.
+                Default to False.
+        """
+        if inference_only and not has_inference_model:
+            raise ValueError
+        self.has_inference_model = has_inference_model
+        self.Inference_only = inference_only
+
+    _inference_model: InferenceModel | None = None
+
+    @property
+    def inference_model(self) -> InferenceModel:
+        """Get inference model.
+
+        Returns:
+            InferenceModel: A model to infer.
+        """
```

Comment:
`property`の`getter`は戻り値情報が自明なので簡潔に。
```suggestion
    def inference_model(self) -> InferenceModel:
        """Get inference model."""
```